### PR TITLE
remove Legacy Numeric builtins for the compiler

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs [hunk 1]
```diff
@@ -186,7 +186,6 @@ featureExperimental = Feature
 foreverCppFlags :: [T.Text]
 foreverCppFlags =
     [ "DAML_NAT_SYN"
-    , "DAML_NAT_TYPE_ERASURE"
```
Comment (paulbrauner-da) [inline compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs:189]: I think we want to keep it: the reason we have these flags is in (the admittedly unlikely) case someone was relying on these CPP flags being set in their daml file. That way we don't break backwards source compatibility.

File: compiler/damlc/daml-stdlib-src/DA/Internal/Prelude.daml [hunk 2]
```diff
@@ -568,11 +568,7 @@ truncate = primitive @"BENumericToInt64"
 
 -- | Convert an `Int` to a `Numeric`.
 intToNumeric : forall n. NumericScale n => Int -> Numeric n
-#ifndef DAML_NAT_TYPE_ERASURE
```
Comment (paulbrauner-da) [inline compiler/damlc/daml-stdlib-src/DA/Internal/Prelude.daml:571]: Looks like I had missed this one and the ones below!

**Review by paulbrauner-da — APPROVED** [review]
LGTM. Please keep the flat in the foreverFlags.

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs (hunk 1 above)
Comment (remyhaemmerle-da) [reply compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs:189]: This feature was never made stable, so nobody should rely on it.

**Pull request merged**